src/components/subscriptions/data/hooks.js

```diff
@@ -0,0 +1,159 @@
+ * It is also dependent on the search query state provided by SubscriptionDetailContext.
+ */
+export const useSubscriptionUsersOverview = ({
+  subscriptionUUID,
+  search,
+  errors,
+  setErrors,
+}) => {
+  const initialSubscriptionUsersOverview = {
+    all: 0,
+    activated: 0,
+    assigned: 0,
+    revoked: 0,
+  };
+  const [subscriptionUsersOverview, setSubscriptionUsersOverview] = useState(initialSubscriptionUsersOverview);
+
+  useEffect(() => {
+    if (subscriptionUUID) {
+      LicenseManagerApiService.fetchSubscriptionUsersOverview(subscriptionUUID, { search })
+        .then((response) => {
+          const subscriptionUsersOverviewData = response.data.reduce((accumulator, currentValue) => ({
+            ...accumulator, [currentValue.status]: currentValue.count,
+          }), initialSubscriptionUsersOverview);
+          subscriptionUsersOverviewData.all = response.data.reduce(
+            (accumulator, currentValue) => accumulator + +currentValue.count,
+            0,
+          );
+          setSubscriptionUsersOverview(camelCaseObject(subscriptionUsersOverviewData));
+        })
+        .catch((err) => {
+          NewRelicService.logAPIErrorResponse(err);
+          setErrors({
+            ...errors,
+            [SUBSCRIPTION_USERS_OVERVIEW]: NETWORK_ERROR_MESSAGE,
+          });
+        });
+    }
+  }, [subscriptionUUID, search]);
+
+  return subscriptionUsersOverview;
+};
+
+/*
+ * This hook provides a list of users for a given subscription UUID.
+ * It is also dependent on state from SubscriptionDetailContext.
+ */
+export const useSubscriptionUsers = ({
+  activeTab,
+  currentPage,
+  searchQuery,
+  subscriptionUUID,
+  errors,
+  setErrors,
+}) => {
+  const [subscriptionUsers, setSubscriptionUsers] = useState({
+    results: [],
+    count: 0,
+    next: null,
+    previous: null,
+  });
+
+  useEffect(() => {
+    if (!subscriptionUUID) {
+      return;
+    }
+    const options = {
+      status: licenseStatusByTab[activeTab],
+      search: searchQuery,
+      page: currentPage,
+    };
+    LicenseManagerApiService.fetchSubscriptionUsers(subscriptionUUID, options)
+      .then((response) => {
+        setSubscriptionUsers(camelCaseObject(response.data));
+      })
+      .catch((err) => {
+        NewRelicService.logAPIErrorResponse(err);
+        setErrors({
+          ...errors,
+          [SUBSCRIPTION_USERS]: NETWORK_ERROR_MESSAGE,
+        });
+      });
+  }, [activeTab, currentPage, searchQuery, subscriptionUUID]);
+
+  return subscriptionUsers;
+};
+
+/*
+ * This hook provides top level subscription data for the given enterprise customer UUID.
+ * It also provides an error state to be used by all subscription and license components.
+*/
+export const useSubscriptionData = ({ enterpriseId }) => {
+  const [errors, setErrors] = useState({});
+  const {
+    subscriptions,
+    forceRefresh,
+  } = useSubscriptions({ enterpriseId, errors, setErrors });
+
+  return {
+    subscriptions,
```

Comment:
really nitty: maybe group subscriptions and forceRefresh together? to separate out subscriptions data vs the useState outputs?

Reply:
I'm inclined to keep this as is and avoid adding another level of nesting